src/client/dfs/dfs.c

```diff
@@ -1849,7 +1838,47 @@ dfs_lookup(dfs_t *dfs, const char *path, int flags, dfs_obj_t **_obj,
 	for (token = strtok_r(rem, "/", &sptr);
 	     token != NULL;
 	     token = strtok_r(NULL, "/", &sptr)) {
-dfs_lookup_loop:
+lookup_rel_path_loop:
+
+		/*
+		 * Open the directory object one level up.
+		 * Since fetch_entry does not support ".",
+		 * we can't support ".." as the last entry,
+		 * nor can we support "../.." because we don't
+		 * have parent.parent_oid and parent.mode.
+		 * For now, represent this partial state with
+		 * parent_fully_valid.
```

Comment:
i have some thought on how to support .., but unfortunately it required a layout change and some modifications to rename. but something for the future

Reply:
Sure. It's an edge case anyway